internal/k8s/builder/gateway.go

```diff
@@ -149,18 +150,34 @@ func (b *GatewayDeploymentBuilder) Build() *v1.Deployment {
 	}
 }
 
+func (b *GatewayDeploymentBuilder) instances() *int32 {
```

Comment:
It looks like that might already be getting preserved since it looks like replicas is a layer above template. Might need you to double check on that though.

Reply:
Ah, interesting, yeah, that would make sense, that's definitely not intended. We should probably audit what isn't handled by this merge operation apart from metadata (which was IIRC the intent of adding this create/update code that bypasses things like metadata).

Think best way to double check this and also make sure we're always allowing for manual modification is to probably add an e2e test for this.

Reply:
Added two e2e tests, one for normal replication and one for testing to make sure we can modify the replica number manually. Let me know your thoughts.